### PR TITLE
Add changelog entry about new `allow_other_host` option for `redirect_back` method

### PR DESCRIPTION
**Pull request by bogdanvlviv**
Related to #30850

### CODE REVIEW

File: actionpack/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+*   Add `:allow_other_host` option to `redirect_back` method.
+    When `allow_other_host: false` redirection back to URLs from
+    a different host is not allowed. `allow_other_host` is `true` by default.
```
Comment (eileencodes) [inline actionpack/CHANGELOG.md]: I'm having trouble parsing what this sentence is saying "When `allow_other_host: false` redirection back to URLs from a different host is not allowed". Should it say something more like "When `allow_other_host` is set to false, the `redirect_back` URL will now allow a redirecting from a different host."

Comment (bogdanvlviv) [reply actionpack/CHANGELOG.md]: What about:
"When `allow_other_host` is set to `false`, the `redirect_back` will not allow a redirecting to a different host."

Comment (eileencodes) [reply actionpack/CHANGELOG.md]: Yup that's better.

Comment (bogdanvlviv) [reply actionpack/CHANGELOG.md]: Amended

**Pull request merged**